src/sctools/metrics/merge.py

```diff
@@ -38,14 +40,87 @@ class MergeGeneMetrics(MergeMetrics):
     # todo this class needs to do a weighted mean; this is something numpy can do.
     def execute(self) -> None:
         """
-        Currently this class returns an incorrect result
+        Genes are expected to be found in multiple shards. Thus, the metric values must be
+        summed or averaged over, depending on how they were originally calculated.
+
+        This function exectures the merge and writes the resulting data to self.output_file
         """
-        metric_dataframes: List[pd.DataFrame] = [
-            pd.read_csv(f, index_col=0) for f in self._metric_files
+
+        # different columns in the gene dataset need to be merged in different ways. Count data
+        # must be summed, while averages need to have a (weighted) average taken
+
+        columns_to_sum = [
+            'n_reads',
+            'noise_reads',
+            'perfect_molecule_barcodes',
+            'reads_mapped_exonic',
+            'reads_mapped_intronic',
+            'reads_mapped_utr',
+            'reads_mapped_uniquely',
+            'reads_mapped_multiple',
+            'duplicate_reads',
+            'spliced_reads',
+            'antisense_reads',
+            'n_molecules',
+            'n_fragments',
+            'fragments_with_single_read_evidence',
+            'molecules_with_single_read_evidence',
+            'number_cells_detected_multiple',
+            'number_cells_expressing',
         ]
-        index: pd.Index = metric_dataframes[0].index
-        columns: pd.Index = metric_dataframes[0].columns
-        averaged_data: np.ndarray = np.mean(metric_dataframes, axis=0)
+
+        sum_operations = {c: 'sum' for c in columns_to_sum}
+
+        def weighted_average(x):
```

Comment:
How about naming this `data_frame` instead of `x` for improved readability?

Reply:
Done.